src/Conversion/ONNXToKrnl/Tensor/Split.cpp

```diff
@@ -45,40 +46,45 @@ LogicalResult ONNXSplitOpLoweringCommon(Operation *op, ArrayRef<Value> operands,
   }
 
   // Creates loops, one for each output.
-  for (unsigned int i = 0; i < outputNum; ++i) {
+  for (unsigned i = 0; i < outputNum; ++i) {
     OpBuilder::InsertionGuard insertGuard(rewriter);
-    // Create loop.
-    BuildKrnlLoop outputLoops(rewriter, loc, rank);
-    outputLoops.createDefineAndIterateOp(allocs[i]);
-    rewriter.setInsertionPointToStart(outputLoops.getIterateBlock());
 
     // Scope for krnl ops
     IndexExprScope childScope(&rewriter, shapeHelper.scope);
+
     KrnlBuilder createKrnl(rewriter, loc);
+    ValueRange loopDef = createKrnl.defineLoops(rank);
+    SmallVector<IndexExpr, 4> lbs(rank, LiteralIndexExpr(0));
+
+    MemRefBoundsIndexCapture allocsBounds(allocs[i]);
+    SmallVector<IndexExpr, 4> ubs;
+    for (uint64_t i = 0; i < rank; ++i)
+      ubs.emplace_back(allocsBounds.getDim(i));
```

Comment:
This loop can be replaced by `allocsBounds.getList(ubs)`.

Reply:
Thanks for pointing that out. Will use `dataBounds.getDimList(ubs)` because `getList` is private.